assembly/std/solidity_compat/misc.ts

```diff
@@ -10,3 +15,25 @@ import { wrapStaticArray } from '@massalabs/as-types';
 export function bytes32ToU256(a: StaticArray<u8>): u256 {
   return u256.fromUint8ArrayBE(wrapStaticArray(a));
 }
+
+// implementation Solidity selfdestruct:
+// https://www.infuy.com/blog/using-self-destruct-in-solidity-contracts/
+export function selfDestruct(transferToAddr: Address): void {
+  // 1- empty the SC
+  let emptySc = new StaticArray<u8>(0);
+  env.setBytecode(emptySc);
+
+  // 2- delete everything in Storage
+  let keys = env.getKeys(new StaticArray<u8>(0));
+  for (let i = 0; i < keys.length; i++) {
+    Storage.del(keys[i]);
+  }
```

Comment:
If to many Keys I think this function will fail. Could we add a system to batch key removal.

Reply:
I've added a small note for developers about that :)